### PR TITLE
Make less web3 calls

### PR DESCRIPTION
**Pull request by sparrowDom**
In this PR: 
- make only 1 getArp call on dapp init
- make balances & allowances calls only when user is active in the dapp

related issue: https://github.com/OriginProtocol/origin-dollar/issues/240#issuecomment-698129245

### CODE REVIEW

File: dapp/src/components/AccountListener.js [hunk 1]
```diff
@@ -14,7 +14,10 @@ const AccountListener = (props) => {
   const web3react = useWeb3React()
   const { account, chainId, library } = web3react
   const prevAccount = usePrevious(account)
+  const [contracts, setContracts] = useState(null)
+  const [balancesInterval, setBalancesInterval] = useState(null)
```
Comment (shahthepro) [inline dapp/src/components/AccountListener.js:18]: Any reason to have it as a state and not a local variable inside `useEffect`?

File: dapp/src/components/UserActivityListener.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      })
+      this.setState({ userActivityState: 'active' })
+    }
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousemove', this.onMouseMove)
+    // check for user activity and adjust states when needed
+    setInterval(this.onStateCheck, 500)
+  }
```
Comment (shahthepro) [inline dapp/src/components/UserActivityListener.js:48]: Cleanup and remove event listeners on `componentWillUnmount`

**Review by shahthepro — COMMENTED** [review]
A few comments inline

File: dapp/src/components/UserActivityListener.js (hunk 2 above)
Comment (sparrowDom) [reply dapp/src/components/UserActivityListener.js:48]: yup god point thanks

File: dapp/src/components/AccountListener.js (hunk 1 above)
Comment (sparrowDom) [reply dapp/src/components/AccountListener.js]: Yup your right cleaning that up

**Comment by sparrowDom** [comment]
thanks @shahthepro both comments addressed. If nothing else we are good to merge this one I think?

**Review by shahthepro — APPROVED** [review]
Everything else looks good to me. LGTM

**Pull request merged**